File: getting-started/src/main/java/demo/gettingstarted/persistence/jpa/entitygraph/repository/UserRepository.java
package demo.gettingstarted.persistence.jpa.entitygraph.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import demo.gettingstarted.persistence.jpa.entitygraph.entity.User;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
}

File: JPA/jpa-study/src/main/kotlin/com/demo/jpastudy/repository/UserRepository.kt
package com.demo.jpastudy.repository

import com.demo.jpastudy.entity.AddressEntity
import com.demo.jpastudy.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<UserEntity, Long> {
    fun findByName(name: String): UserEntity?
}
interface AddressRepository : JpaRepository<AddressEntity, Long> {
    fun findByAddress(address: String): AddressEntity
}
